/*
 * Specification for DAQ configuration.
 */
syntax = "proto3";

/*
 * DAQ configuration.
 */
message ForchConfig {
  // Description description for dashboard.
  string site_description = 1;

  // Default time to monitor before starting tests.
  int32 monitor_scan_sec = 2;

  // default timeout for tests, or 0 to disable.
  int32 default_timeout_sec = 3;

  // Default base module configuration file.
  string base_conf = 4;

  // Directory of site-configuration data, including devices.
  string site_pathk = 5;

  // initial dhcp lease time on test run start
  string initial_dhcp_lease_time = 6;

  // dhcp lease time after initial startup
  string dhcp_lease_time = 7;

  // dhcp response time for testing long test cycle
  string long_dhcp_response_sec = 8;

  // Interface name for switch control plane  
  string ext_ctrl = 9;

  // IP address and subnet for control plane interface
  string ext_ofip = 10;

  // IP address of external switch.
  string ext_addr = 11;
  
  // IP address template and subnet for module ip addresses
  string ext_loip = 12;

  // Upstream dataplane port from the external (secondary) switch.
  int32 sec_port = 13;

  // CSV separated list of names to assign to external switch interfaces.
  string intf_names = 14;

  // Command file with startup commands.
  string startup_cmds = 15;

  // Specify explicit host test configuration.
  string host_tests = 16;

  // Limit the number of test runs before exit.
  int32 run_limit = 17;

  // Stop after there are any errors.
  bool fail_mode = 18;

  // Keep results around and don't kill containers on error.
  bool result_linger = 19;

  // Don't run tests on devices, just activate them on the network.
  bool no_test = 20;

  // Extra debugging arguments. Can be one of { error, warn, info, debug }
  string daq_loglevel = 21;

  // Mininet debugging arguments. Can be one of { error, warn, info, debug }
  string mininet_loglevel = 22;

  // Service account credential for access to a cloud project.
  string gcp_cred = 23;

  // Topic used for live stream validation.
  string gcp_topic = 24;

  // Validation schema default.
  string schema_path = 25;

  // Location of mud files to use for this setup.
  string mud_files = 26;

  // File to use for runtime device behavior specification.
  string device_specs = 27;

  // Configuration directory for runtime tests.
  string test_config = 28;

  // Set port-debounce for flaky connecitons. Zero to disable.
  int32 port_debounce_sec = 29;

  // Hook for failure diagnostics.
  string fail_hook = 30;

  // Directory of defaults for new devices.
  string device_template = 31;

  // If defined, override destination root for device reports (defaules to {site_path})
  string site_reports = 32;

  // Cloud data retention policy.
  float run_data_retention_days = 33;
}
